src/controllers/hid/hidcontroller.cpp

```diff
@@ -277,6 +277,38 @@ void HidController::send(QByteArray data, unsigned int reportID) {
     data.prepend(reportID);
 
     int result = hid_write(m_pHidDevice, (unsigned char*)data.constData(), data.size());
+    if (result == -1) {
+        if (ControllerDebug::enabled()) {
+            qWarning() << "Unable to send data to" << getName()
+                       << "serial #" << hid_serial << ":"
+                       << safeDecodeWideString(hid_error(m_pHidDevice), 512);
+        } else {
+            qWarning() << "Unable to send data to" << getName() << ":"
+                       << safeDecodeWideString(hid_error(m_pHidDevice), 512);
+        }
+    } else {
+        controllerDebug(result << "bytes sent to" << getName()
+                               << "serial #" << hid_serial
+                               << "(including report ID of" << reportID << ")");
+    }
+}
+
+void HidController::send_feature_report(
+        QList<int> data, unsigned int length, unsigned int reportID) {
+    Q_UNUSED(length);
+    QByteArray temp;
+    foreach (int datum, data) {
+        temp.append(datum);
+    }
+    send_feature_report(temp, reportID);
+}
+
+void HidController::send_feature_report(QByteArray data, unsigned int reportID) {
+    // Append the Report ID to the beginning of data[] per the API..
+    data.prepend(reportID);
```

Comment:
Won't this modify the original data array?

Reply:
I copied this code from the very similar method HidController::send and it sent what I expected. See the screenshot of the USB communication above. Now this data array is an internal variable of the method, therefore it shouldn't harm any code outside.